cumulusci/tasks/bulkdata/utils.py

```diff
@@ -13,15 +12,66 @@
 from cumulusci.utils import convert_to_snake_case
 
 
-def batch_iterator(iterator, n=10000):
-    while True:
-        batch = list(itertools.islice(iterator, n))
-        if not batch:
-            return
+def batch_iterator(iterator, fields, n=10000, char_limit=10000000):
```

Comment:
I don't love that we have to pull in fields here but didn't see a way around it.